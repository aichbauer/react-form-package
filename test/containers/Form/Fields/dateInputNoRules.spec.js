import React from 'react';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { configure, mount } from 'enzyme';

import {
  Form,
  Field,
  Button,
} from '../../../../src';
import { dateInputNoRules } from '../../../fixtures';

configure({ adapter: new Adapter() });

test('Render Form with date input | date input no rules', () => {
  const myComponent = (
    <Form>
      <div>
        <div>date</div>
        <Field type="date" id="date" />
      </div>
      <div>
        <Button id="submit" type="submit" onClick={(state) => state}>submit</Button>
      </div>
    </Form>
  );

  const tree = renderer.create(myComponent).toJSON();
  const wrapper = mount(myComponent);
  const state = wrapper.state();

  expect(state).toEqual(dateInputNoRules);
  expect(tree).toMatchSnapshot();
});
