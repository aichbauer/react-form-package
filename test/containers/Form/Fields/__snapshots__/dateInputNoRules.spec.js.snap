// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form with date input | date input no rules 1`] = `
<form
  enctype="application/x-www-form-urlencoded"
>
  <div>
    <div>
      date
    </div>
    <input
      className="rfp-input"
      id="date"
      invalid={false}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="date"
      valid={false}
      value=""
    />
  </div>
  <div>
    <button
      className="rfp-button"
      disabled={false}
      id="submit"
      onClick={[Function]}
      type="submit"
    >
      submit
    </button>
  </div>
</form>
`;
